collective/soupstrainer/__init__.py

```diff
@@ -33,10 +33,10 @@ def __init__(self, exclusions=None, style_whitelist=None,
         self.class_blacklist = set(class_blacklist)
 
     def __call__(self, data):
-        if isinstance(data, basestring):
-            return unicode(self.clean(bs4.BeautifulSoup(data, self.parser)))
-        else:
+        if isinstance(data, bs4.BeautifulSoup):
             return self.clean(data)
+        else:
+            return str(self.clean(bs4.BeautifulSoup(data, self.parser)))
```

Comment:
What about `six.text_type` instead of `str`? This way it is "unicode"  on all Python versions and the API does not change.